LabelsRandomColors2.0/src/com/company/Main.java

```diff
@@ -61,21 +61,20 @@ public void mousePressed(MouseEvent e) {
             /** in debugging, quando arriva alla riga successiva, esce dal debugging, ma continua
              * ad eseguire correttamente il programma
              *
+             *
              * Ho provato a sostituire in setColor un colore fisso (es ROSSO) per vedere se il resto del codice funzia,
              * ed ha esito positivo. Quindi c'e qualcosa con la istruzione "decode" che non va.
              * nessun errore, nessuna warnings
              */
 
+
             gobjNew.setColor(Color.decode(gobjNew.getLabel()));

```

Comment:
Per aiutarti a capire cosa va storto con `Color.decode`, potrsti navigare alla sua definizione usando  'Go to declaration' in IntelliJ (https://www.jetbrains.com/help/rider/Navigation_and_Search__Go_to_Declaration.html)

Reply:
public static Color decode(String nm) throws NumberFormatException {
        Integer intval = Integer.decode(nm);
        int i = intval.intValue();
        return new Color((i >> 16) & 0xFF, (i >> 8) & 0xFF, i & 0xFF);
    }

Mi sa che lo interpreto male, perche leggendo il codice sopra e seguendo la logica di quello che ho imparato fin ora, mi verrebbe da dire che la chiamata di decode diventa un riferimento circolare ( decode ha dentro Integer.decode richiama decode, che ha dentro Integer.decode....) ma immagino non sia cosi.

Ho guardato anche la definizione di Integer decode e la mia interpretazione di "cosa e' andato storto e:
Acquisendo come parametro la stringa "RED" o "YELLOW" etc, una volta convertita in Int tale stringa, il valore in Int non corrisponde a nessun colore di quelli possibili)
( perche' la condizione ---> Color((i >> 16) & 0xFF, (i >> 8) & 0xFF, i & 0xFF) <-- in decode, non viene soddisfatta)